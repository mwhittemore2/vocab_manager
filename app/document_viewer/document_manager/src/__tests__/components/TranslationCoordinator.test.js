import React from "react"
import { shallow } from "enzyme"
import toJSON from 'enzyme-to-json'
import { EndpointManager, Punctuation, QueueManager,
         Spacing, TextManager, TranslationCoordinator } from '../../components/ui/TranslationCoordinator'
import C from '../../constants'

describe("EndpointManager", () => {
    let setEndpoint
    beforeEach(() => {
        setEndpoint = jest.fn()
    })

    afterAll(() => {
        jest.resetAllMocks()
    })
    
    it("check tags", () => {
        let txt = toJSON(
                      shallow(<EndpointManager setEndpoint={setEndpoint}/>)
                      .find('div')
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("QueueManager", () => {
    let clearQueue
    beforeEach(() => {
        clearQueue = jest.fn()
    })

    afterAll(() => {
        jest.resetAllMocks()
    })

    it("check tags", () => {
        let txt = toJSON(
                      shallow(<QueueManager clearQueue={clearQueue}/>)
                      .find('div')
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("Punctuation", () => {
    let addText
    beforeEach(() => {
        addText = jest.fn()
    })

    afterAll(() => {
        jest.resetAllMocks()
    })

    it("add comma button", () => {
        let punctuation = ","
        let txt = toJSON(
                      shallow(<Punctuation addText={addText} 
                                           punct={punctuation}/>
                      )
                      .find('button')
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("Spacing", () => {
    let addText
    beforeEach(() => {
        addText = jest.fn()
    })

    afterAll(() => {
        jest.resetAllMocks()
    })

    it("add whitespace button", () => {
        let description = "whitespace"
        let spacing = ' '
        let txt = toJSON(
                      shallow(<Spacing addText={addText}
                                       description={description}
                                       spacing={spacing}/>
                      )
                      .find('button')
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("TextManager", () => {
    let addText
    beforeEach(() => {
        addText = jest.fn()
    })

    afterAll(() => {
        jest.resetAllMocks()
    })

    it("check tags", () => {
        let txt = toJSON(
                      shallow(<TextManager addText={addText}/>)
                      .find('div')
                  )
        expect(txt).toMatchSnapshot()
    })
})

describe("TranslationCoordinator", () => {
    let addText
    let clearQueue
    let interaction
    let option
    let setEndpoint
    let translate
    beforeEach(() => {
        addText = jest.fn()
        clearQueue = jest.fn()
        interaction = C.DOCUMENT_VIEWER
        option = C.DOC_VIEWER_OPTIONS.TRANSLATION_COORDINATOR
        setEndpoint = jest.fn()
        translate = jest.fn()
    })
    it("check tags", () => {
        let txt = toJSON(
                      shallow(<TranslationCoordinator addText={addText}
                                                      clearQueue={clearQueue}
                                                      interaction={interaction}
                                                      option={option}
                                                      setEndpoint={setEndpoint}
                                                      translate={translate}/>
                      )
                      .find('div')
                  )
        expect(txt).toMatchSnapshot()
    })
})