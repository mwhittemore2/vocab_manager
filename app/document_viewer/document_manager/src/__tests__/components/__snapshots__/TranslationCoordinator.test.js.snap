// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndpointManager check tags 1`] = `
<div
  id="translation-coordinator-endpoint-manager"
>
  Set the boundaries of the text to be translated
  <br />
  <button>
    Start
  </button>
  <button>
    Finish
  </button>
</div>
`;

exports[`Punctuation add comma button 1`] = `
<button>
  ,
</button>
`;

exports[`QueueManager check tags 1`] = `
<div
  id="translation-coordinator-queue-manager"
>
  Control the content of the translation queue
  <br />
  <button />
</div>
`;

exports[`Spacing add whitespace button 1`] = `
<button>
  whitespace
</button>
`;

exports[`TextManager check tags 1`] = `
<div
  id="translation-coordinator-text-manager"
>
  Select the text to be added to the translation queue.
  <br />
  Punctuation
  <br />
  <Punctuation
    addText={[MockFunction]}
    key="text-manager-0-,"
    punct=","
  />
  <Punctuation
    addText={[MockFunction]}
    key="text-manager-1-."
    punct="."
  />
  <Punctuation
    addText={[MockFunction]}
    key="text-manager-2-'"
    punct="'"
  />
  <Punctuation
    addText={[MockFunction]}
    key="text-manager-3-\\""
    punct="\\""
  />
  <br />
  Spacing
  <br />
  <Spacing
    addText={[MockFunction]}
    description="whitespace"
    key="text-manager-0-whitespace"
    spacing=" "
  />
  <br />
  Custom Text
</div>
`;

exports[`TranslationCoordinator check tags 1`] = `
<div
  id="translation-coordinator"
>
  <TextManager
    addText={[MockFunction]}
  />
  <EndpointManager
    setEndpoint={[MockFunction]}
  />
  <QueueManager
    clearQueue={[MockFunction]}
  />
  <button>
    Get Translations
  </button>
</div>
`;
